src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -84,17 +84,36 @@ public static final int compareLocatables(Locatable first, Locatable second, SAM
             // compare the contigs
             result = compareContigs(first, second, dictionary);
             if (result == 0) {
-                // compare start position
-                result = Integer.compare(first.getStart(), second.getStart());
-                if (result == 0) {
-                    // compare end position
-                    result = Integer.compare(first.getEnd(), second.getEnd());
-                }
+                result = compareLocatablesOfSameContig(first, second);
             }
         }
         return result;
     }
 
+    /**
+     * Compare two locations assuming they are of the same contig.
+     *
+     * @return 0 if first and second are equal, a negative value if first < second or a positive value if first > second,
+     * @throws IllegalArgumentException if two locatables are not of the same contig
+     */
+    public static final int compareLocatablesOfSameContig(final Locatable first, final Locatable second) {
```

Comment:
@SHuang-Broad What's the need for this method?  Are you seeing that checking the contigs for equality is expensive somewhere in your code?

Reply:
I was trying to avoid hauling around sequence dictionary around, which the general method requires.